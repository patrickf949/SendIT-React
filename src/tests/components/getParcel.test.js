import React from "react";
import { shallow } from "enzyme";
import {
  mapStateToProps,
  GetParcel as OneParcel
} from "../../components/getParcel";
import { data } from "../mock_data/data";

describe("GetParcel Component", () => {
  const props = {
    match: {
      params: {
        id: "21"
      }
    },
    getParcelAction: jest.fn(),

    parcel: data.parcel
  };
  const props1 = props;
  props1.parcel = { message: "no parcel" };
  it("should render without crashing", () => {
    const wrapper = shallow(<OneParcel {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render without crashing if parcel don't exist", () => {
    const wrapper = shallow(<OneParcel {...props1} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should tests mapstateToprops", () => {
    const state = {
      get_parcel: { parcel: data.parcel }
    };
    expect(mapStateToProps(state)).toEqual({
      parcel: data.parcel
    });
  });
  it("should receive props", () => {
    const wrapper = shallow(<OneParcel {...props} />);

    wrapper.instance().componentDidMount();
    expect(wrapper).toMatchSnapshot();
  });
});
