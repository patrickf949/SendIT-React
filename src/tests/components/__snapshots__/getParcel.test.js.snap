// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetParcel Component should receive props 1`] = `ShallowWrapper {}`;

exports[`GetParcel Component should render without crashing 1`] = `ShallowWrapper {}`;

exports[`GetParcel Component should render without crashing if parcel don't exist 1`] = `ShallowWrapper {}`;
